extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -140,19 +141,29 @@ private FileSystem initializeFileSystem() {
         }
     }
 
-    public Map<String, String> getExtensionArtifacts(final String extensionName) throws StoreAccessException {
+    public Map<String, String> getExtensionArtifacts(final String extensionName) throws
+             FalconException {
         Map<String, String> extensionFileMap = new HashMap<>();
+        Path extensionPath;
         try {
-            Path extensionPath = new Path(storePath, extensionName.toLowerCase());
-            RemoteIterator<LocatedFileStatus> fileStatusListIterator = fs.listFiles(extensionPath, true);
+            RemoteIterator<LocatedFileStatus> fileStatusListIterator;
+            if (AbstractExtension.isExtensionTrusted(extensionName)){
```

Comment:
I looked at the ExtensionService again:
`List<String> extensions = getExtensions();
            for (String extension : extensions) {`
This list of extensions is only be trusted, so, won't need that check there. isn't it?